runtime/interpreter/interpreter.go

```diff
@@ -637,7 +637,7 @@ func (interpreter *Interpreter) RecoverErrors(onError func(error)) {
 	}
 }
 
-func (interpreter *Interpreter) VisitProgram(program *ast.Program) ast.Repr {
+func (interpreter *Interpreter) VisitProgram(program *ast.Program) any {
```

Comment:
Why are we using `any` here? Can we have these interpreter functions return `Value` like the checker ones return `Type`?

Reply:
Unfortunately there are currently two types being returned, `Value` and `controlReturn`. 

That is why for now I kept the interpreter code as-is (`ast.Repr` was an alias for `any`). 
In a subsequent PR I was planning on refactoring this to a single "enum" `Result`, which can represent both.

Reply:
Continued the improvements in #1919